src/library/baseplaylistfeature.cpp

```diff
@@ -478,6 +483,102 @@ void BasePlaylistFeature::slotExportPlaylist() {
     }
 }
 
+void BasePlaylistFeature::slotExportTrackFiles() {
+    // TODO: This is a near-exact duplication of the same code in CreateFeature.
+    // Factor it out!
```

Comment:
> what's a good place to put factored-out code like this?

I think you're going to end up needing a few different components -- a dialog, a thread for doing the work in, and such so maybe library/export/ is a reasonable place to stick this code? The LibraryScanner is a similar model.